coulomb/js/src/main/scala/coulomb/typename.scala

```diff
@@ -0,0 +1,54 @@
+package coulomb
+
+import scala.language.experimental.macros
+
+import scala.reflect.macros.whitebox
+
+import scala.reflect.macros.blackbox.Context
+
+trait UnitTypeName[T] {
+  def name: String // typeName
+  def typeString: String
+  def =:=[U](t: UnitTypeName[U]): Boolean
+}
+
+object UnitTypeName {
+  @inline
+  def apply[T](implicit ev: UnitTypeName[T]): UnitTypeName[T] = ev
+
+  /**
+    * Build an instance of `UnitTypeName` for the type `A`
+    */
+  def unitTypeNameImpl[A: c.WeakTypeTag](
```

Comment:
This macro is partially able to create a new instance of `UnitName` reading the type name from the AST

Reply:
What is your opinion on doing it with the macro, versus using `ClassTag`?   I'm fine with using a macro, mostly just curious.

Reply:
Basically I did the macro as it was suggested initially. Maybe I can try to do it with `ClassTag` and if all tests pass call it done? It was nice to learn a bit about macros but maybe it is a problem long term. It may induce the creation of a lot of code too but I'm not sure about that